### PR TITLE
UCT/IB: Don't fail if no devices are found; return an empty list

### LINKED ISSUE #3611: UCX: standalone machine issue, can't find device
On a standalone (+ tcp, knem, xpmem) Fedora 30 x86_64 machine, the following program works with UCX 1.5.1 but fails with github master, which has ucp_init() returning "No such device"

master @ d4cc8b1fc6ccb220d806b3e888141ea8adb92f30

```C
#include <stdio.h>

#include <ucp/api/ucp.h>

ucp_config_t *cfg;
ucp_context_h ctxt;

int
main()
{
    ucs_status_t s;
    ucp_params_t pm;

    s = ucp_config_read(NULL, NULL, &cfg);
    fprintf(stderr, "ucp_config_read(): %s\n", ucs_status_string(s));

    pm.field_mask = UCP_PARAM_FIELD_FEATURES;
    pm.features   = UCP_FEATURE_RMA;

    s = ucp_init(&pm, cfg, &ctxt);
    fprintf(stderr, "ucp_init(): %s\n", ucs_status_string(s));

    fprintf(stderr, "DEBUG: ctxt = %p\n", ctxt);

    ucp_config_release(cfg);

    return 0;
}
```

```shell
ucp_config_read(): Success
ucp_init(): No such device
DEBUG: ctxt = (nil)
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #3611 @tonycurtis

### CODE REVIEW

File: src/uct/ib/base/ib_md.c [hunk 1]
```diff
@@ -1239,8 +1239,7 @@ static ucs_status_t uct_ib_query_md_resources(uct_md_resource_desc_t **resources
     device_list = ibv_get_device_list(&num_devices);
     if (device_list == NULL) {
         ucs_debug("Failed to get IB device list, assuming no devices are present");
-        status = UCS_ERR_NO_DEVICE;
-        goto out;
+        num_devices = 0;
     }
 
     resources = ucs_calloc(num_devices, sizeof(*resources), "ib resources");
```
Comment (brminich) [inline src/uct/ib/base/ib_md.c]: can calloc return `NULL` now? If yes, the func will fail with `UCS_ERR_NO_MEMORY`

Comment (dmitrygladkov) [inline src/uct/ib/base/ib_md.c]: look like, it can be NULL on some systems when alloc 0 size buffer

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10330/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10332/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_md.c [hunk 2]
```diff
@@ -1258,8 +1259,9 @@ static ucs_status_t uct_ib_query_md_resources(uct_md_resource_desc_t **resources
     status = UCS_OK;
 
 out_free_device_list:
-    ibv_free_device_list(device_list);
-out:
+    if (device_list != NULL) {
```
Comment (brminich) [inline src/uct/ib/base/ib_md.c]: this check is not needed anymore

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10359/ for details (Mellanox internal link).

**Pull request merged**